api/services/locationService.ts

```diff
@@ -71,7 +71,7 @@ async function getLocations(
     )
   LIMIT ${limit}`;
 
-  const response: ScosAgencyDto[] = await makeSCOSRequest(query);
+  const response = await makeSCOSRequest<ScosAgencyDto[]>(query);
```

Comment:
I am having problems with this change and my changes to locationService.ts.  

```
const response: ScosAgencyDto[] = await makeSCOSRequest(query);
  const promises = response.map(mapToLocationDto);
  const mapped = await Promise.all(promises);
  const filtered = mapped.filter(notEmpty);

  log.debug(`Returning ${filtered.length} locations`);
  return filtered;
```

If I change

```
const response: ScosAgencyDto[] = await makeSCOSRequest(query);
```
to

```
const response = await makeSCOSRequest(query);
```

and compile with 'npm tsc' I get:

Type 'unknown' is not assignable to type 'LocationDto'

It turns out that that makeSCOSRequest returns a promise of <any>.  So I tried to change it to <LocationDto[]>.  However, this caused an error in taxonomyService.ts.:

```
api/services/taxonomyService.ts:15:9 - error TS2322: Type 'ScosAgencyDto[]' is not assignable to type 'Pick<ScosCategoryDto, "subcat">[]'.
  Property 'subcat' is missing in type 'ScosAgencyDto' but required in type 'Pick<ScosCategoryDto, "subcat">'.

const response: Pick<ScosCategoryDto, 'subcat'>[] = await makeSCOSRequest(
```
So, it appears that makeSCOSRequest was designed to take advantage of Promise\<any\>.  Can we defer this refactor until we get the freshtrak stuff working?

Reply:
This was already merged into master - let me check out your branch and see if I can get it working. I think you just need to pass in a generic type to `makeSCOSRequest` so that it knows what will be returned in the promise.

Reply:
Also this PR added a change to `makeSCOSRequest` so that it doesn't return `Promise<any>` anymore. It uses the type that you pass in to the generic type so you can do something like `makeSCOSRequest<string[]>` and it knows you will get back a `Promise<string[]>`. You can pass any type into `makeSCOSRequest` like the other dto's that we have to tell the compiler that that is what will be inside the promise.

Reply:
I have a commit for the freshtrakApiService test and linter changes for my code that I haven't pushed yet.  I haven't merged master into my branch yet so maybe if I do that this will be fixed.  I am not a pro at merging and there are a lot of files in master and in my commits, but I will give it a try.

Reply:
I have done my merge commit with my fixes so far.  Should I wait for you before I push?  My stuff works but still has this line
```
const response: ScosAgencyDto[] = await makeSCOSRequest(query);
```

Reply:
There were only a few conflicts when I rebased on top of master - I was able to get mostly everything working but it looks like mocking the `request-promise` package is going to cause issues...which of course is now a deprecated package and no longer being maintained.

Looks like it only works if you use the `jest.mock('request-promise', [mock implementation])` style to mock out what the function will return. I have no idea why but I wasted too much time on it lol. 

Anyways, I have a branch up that has all the necessary fixes if you want to check it out https://github.com/SCODEMeetup/mofb-api/commit/6c9c94a2da6f4ac95e74d2ae49f6bdc5d3359708

Reply:
Thanks for doing this Mike!  I think I have most of the fixes in my last push but I will go through these and make sure I get all of your additions.

Reply:
I just pushed a merge with mh-agency-api to #91 with a couple of exceptions:

My version of [freshtrakServiceApi.test.ts](https://github.com/SCODEMeetup/mofb-api/blob/beebacafe6e3d3c6192ed20eb8a9434ce956f4c9/test/api/services/freshtrakApiService.test.ts) is not using the mock('promise-request... model.  I started with the same solution you came up with but  I think I like:

```
mockGet = jest.spyOn(request, 'get');
      mockGet.mockResolvedValue(
```
better because it doesn't mock promise-request.  The version that wasn't working originally was the JSON response wasn't getting stringified.

Also, my version of [locationService.ts](https://github.com/SCODEMeetup/mofb-api/blob/beebacafe6e3d3c6192ed20eb8a9434ce956f4c9/api/services/locationService.ts) returns freshtrakData rather than freshTrakData (with the upper case T) because that matches the Dto interface and is what the UI is expecting.

Please feel free to share your thoughts in a review of the #91 push or here.